app/models/pipeline_run_stage.rb

```diff
@@ -310,6 +310,35 @@ def db_load_postprocess
       mysqlimport --replace --local --user=$DB_USERNAME --host=#{rds_host} --password=$DB_PASSWORD --columns=taxid,hit_type,first_byte,last_byte,pipeline_run_id --fields-terminated-by=',' idseq_#{Rails.env} taxon_byteranges;
     `
     _stdout, _stderr, _status = Open3.capture3("rm -f #{downloaded_byteranges_path}")
+    notify_users if notify?
+  end
+
+  def project_id
+    pipeline_run.sample.project_id
+  end
+
+  def notify?
```

Comment:
I think 320-341 should be in the model pipeline_run.rb not in pipeline_run_stage.rb  and it will trigger this when all the pipeline_runs are complete.

Reply:
Ok, moved. Did you want me to attach the action to a different event? I thought it would make the most sense to check whenever a run completes (i.e. when db_load_postprocess happens)